integrationconnectors/managedzone/main.tf

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+data "google_project" "target_project" {
+}
+
+resource "google_project_iam_member" "default" {
+  project = data.google_project.target_project.project_id
+  role    = "roles/dns.peer"
+  member  = "serviceAccount:service-${data.google_project.test_project.number}@gcp-sa-connectors.iam.gserviceaccount.com"
+}
+
+resource "google_compute_network" "default" {
+  project                 = data.google_project.target_project.project_id
+  name                    = "test"
+  auto_create_subnetworks = false
+}
+
+resource "google_dns_managed_zone" "default" {
+  name       = "test-dns"
+  dns_name   = "private.example.com."
```

Comment:
Per discussion, if it's possible to create this DNS managed zone for a domain we do not own, then this is fine. Otherwise, we will need to add test skipping.